les/odr_requests.go

```diff
@@ -164,7 +169,15 @@ func (r *HeaderRequest) Validate(db ethdb.Database, msg *Msg) error {
 		return errInvalidMessageType
 	}
 	headers := msg.Obj.([]*types.Header)
-	if len(headers) != 1 {
+	if len(headers) == 0 && r.isByHash() {
+		// For requests by number, we only send to peers for which we know the block number
```

Comment:
How do we know what range of blocks a peer has?

Reply:
It's based on information we maintain about the peer: https://github.com/celo-org/celo-blockchain/blob/1188111ac106a92c6c9d89c5ff3da265b404a161/les/peer.go#L600-L622 .